### ISSUE TITLE
Port to Webview

### ISSUE THREAD

**Issue by RagibHasin** [mozilla/qbrt#182]
As #181 mentioned that this project is practically abandoned, would it be possible to port this runtime to webview extending webview to Gecko support through Firefox.

**Comment by ghost** [comment]
you can fork it.

**Comment by mykmelez** [comment]
It isn't really possible to port this runtime to the (Android System, presumably) WebView, as this project is closely tied to the Gecko API, so a port to WebView would effectively just be a rewrite from scratch.

Note also this other embedded Gecko option for Android: https://mozilla.github.io/geckoview/.

**Comment by RagibHasin** [comment]
> It isn't really possible to port this runtime to the (Android System, presumably) WebView, as this project is closely tied to the Gecko API, so a port to WebView would effectively just be a rewrite from scratch.

I was actually referring to desktop [webview](https://github.com/webview/webview/) runtime. This webview uses WebKit API internally, and I was looking into making an option to use Gecko API to leverage the Firefox already installed in system.

Any pointer is appreciated.

**Comment by mykmelez** [comment]
Gecko doesn't have an embedding API like the WebKit API (except on Android), so it would be difficult to adapt a project like https://github.com/webview/webview/ to use Gecko in addition to WebKit.

For Gecko on desktop, there are ways to build an application like Firefox that embeds Gecko, but it looks a lot more like "fork Firefox and turn it into a different app" than embedding a rendering engine into an application development library.

Sorry, I don't have better news.

**Comment by RagibHasin** [comment]
It Is there any ongoing initiative to create an embedding API for Gecko or otherwise modularizing Firefox components? An embeddable Gecko would be very useful especially with all the Web technology based desktop apps coming up.

**Comment by ghost** [comment]
> It Is there any ongoing initiative to create an embedding API for Gecko or otherwise modularizing Firefox components? An embeddable Gecko would be very useful especially with all the Web technology based desktop apps coming up.

Specially considering that Chromium is dominating the world: see Electron.js, NW.js, PWA, Chrome itself, Chrome for Android, CEF... Chromium is a really good project but a totally free alternative is :heart:

**Comment by RagibHasin** [comment]
Yes. Especially I love Firefox for its font rendering. My native language is Bangla and it renders on Firefox as crisp as English but that is not the case on Chromium based browsers. It was one of the main reason for my selection of Firefox in 2009 when I moved from Internet Explorer.

**Comment by mykmelez** [comment]
@RagibHasin, I don't know about any such effort, other than GeckoView on Android.
